### ISSUE TITLE
BR-*-05 BR-*-06 BR-*-07 BR-*-09 fail on negative zero

### ISSUE THREAD

**Issue by ojundt** [SimplerInvoicing/validation#22]
Just came across an edge case. There are a couple of PEPPOL BIS Billing rules that fail on a negative zero (`-0.0`), e.g. `BR-AE-09` with `<xsl:when test="../cbc:TaxAmount = 0"/>`. Imho these rules are unnecessary strict and negative zeros should be accepted.

These rules are affected:
```
BR-AE-05
BR-AE-06
BR-AE-07
BR-AE-09
BR-E-05
BR-E-06
BR-E-07
BR-E-09
BR-G-05
BR-G-06
BR-G-07
BR-G-09
BR-IC-09
BR-IC-06
BR-IC-07
BR-IC-05
BR-O-09
BR-Z-09
BR-Z-06
BR-Z-07
BR-Z-05
```

**Comment by remcoros** [comment]
**Just out of curiosity**, what kind of system produces "negative or positive 0"? I know IEEE defined negative and positive zero to be used in floating point arithmetic. But it doesn't really makes sense to me in the context of a (financial) business document.

**Comment by ojundt** [comment]
I came across it during the implementation of a `CreditNote` UBL generator where we had to flip all amounts from positive to negative and the other way around by multiplying them with `-1`. That way `0.0` became `-0.0`.

**Comment by tjeb** [comment]
For some reason, I have always mentally skipped over this one when going through the issues list. So let me start with an apology for that.

It's an interesting one; it would appear that -0.0 is not considered 0 by the expression evaluator, and adding 'or <element> = -0.0' does not fix it. Strangely enough, if you change the check to '= -0.0' it stops working for the value -0 (but not 0).

The fix turns out to be to cast cbc:TaxAmount (or any relevant element) with xs:decimal before doing the comparison.

I have included a fix for this commit 6ffb6bee360c8b63d1b9220f1e15a3a5af6f0f4b

I have also made a pull request upstream for this (at the base validation for EN-16931 repository): https://github.com/ConnectingEurope/eInvoicing-EN16931/pull/227
